sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/SampleTest.java

```diff
@@ -47,228 +45,264 @@
 import org.junit.experimental.categories.Category;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Suite;
 
 /**
  * Tests for Sample transform.
  */
-@RunWith(JUnit4.class)
+@RunWith(Suite.class)
+@Suite.SuiteClasses({
+    SampleTest.PickAnyTest.class,
+    SampleTest.MiscTest.class
+})
 public class SampleTest {
-  static final Integer[] EMPTY = new Integer[] { };
-  static final Integer[] DATA = new Integer[] {1, 2, 3, 4, 5};
-  static final Integer[] REPEATED_DATA = new Integer[] {1, 1, 2, 2, 3, 3, 4, 4, 5, 5};
+  private static final Integer[] EMPTY = new Integer[] { };
+  private static final Integer[] DATA = new Integer[] {1, 2, 3, 4, 5};
+  private static final Integer[] REPEATED_DATA = new Integer[] {1, 1, 2, 2, 3, 3, 4, 4, 5, 5};
 
   /**
-   * Verifies that the result of a Sample operation contains the expected number of elements,
-   * and that those elements are a subset of the items in expected.
+   * Test variations for Sample transform.
    */
-  @SuppressWarnings("rawtypes")
-  public static class VerifyCorrectSample<T extends Comparable>
-      implements SerializableFunction<Iterable<T>, Void> {
-    private T[] expectedValues;
-    private int expectedSize;
-
-    /**
-     * expectedSize is the number of elements that the Sample should contain. expected is the set
-     * of elements that the sample may contain.
-     */
-    @SafeVarargs
-    VerifyCorrectSample(int expectedSize, T... expected) {
-      this.expectedValues = expected;
-      this.expectedSize = expectedSize;
+  @RunWith(Parameterized.class)
+  public static class PickAnyTest {
```

Comment:
`testPickAnyWhenEmpty` was doing:
```java
runPickAnyTest(NO_LINES, 0);
runPickAnyTest(NO_LINES, 1);
```
Which are covered by the data variations we generate:

```java
@Parameterized.Parameters(name = "limit_{0}")
    public static Iterable<Object[]> data() throws IOException {
      return ImmutableList.<Object[]>builder()
          .add(
              new Object[] {
                  0
              },
              new Object[] {
                  1
              },
...
```
Am I missing something?